lib/preprocess.js

```diff
@@ -5,20 +5,21 @@ import { CarReader } from '@ipld/car'
 import { validateBlock } from '@web3-storage/car-block-validator'
 import { recursive as exporter } from 'ipfs-unixfs-exporter'
 import createDebug from 'debug'
+import { RoundData } from './round.js'
 
 const debug = createDebug('spark:preprocess')
 
 export class Measurement {
-  constructor (m) {
-    this.participantAddress = parseParticipantAddress(m.participant_address)
-    this.retrievalResult = getRetrievalResult(m)
-    this.cid = m.cid
-    this.spark_version = m.spark_version
+  constructor (m, pointerize = (v) => v) {
+    this.participantAddress = pointerize(parseParticipantAddress(m.participant_address))
+    this.retrievalResult = pointerize(getRetrievalResult(m))
+    this.cid = pointerize(m.cid)
+    this.spark_version = pointerize(m.spark_version)
     this.fraudAssessment = null
-    this.inet_group = m.inet_group
+    this.inet_group = pointerize(m.inet_group)
     this.finished_at = parseDateTime(m.finished_at)
-    this.provider_address = m.provider_address
-    this.protocol = m.protocol
+    this.provider_address = pointerize(m.provider_address)
```

Comment:
Whoa! I forgot to pointerize this field in my original version. After adding this line, the memory usage is down to 30% of the previous version 🎉 🎊